Mathport/Syntax/Translate/Basic.lean

```diff
@@ -365,65 +476,65 @@ def optTy (ty : Option Syntax) : M (Option Syntax) :=
 
 def trCalcArgs (args : Array (Spanned Expr × Spanned Expr)) : M (Array Syntax) :=
   args.mapM fun (lhs, rhs) =>
-    return mkNode ``calcStep #[← trExpr lhs.kind, mkAtom ":=", ← trExpr rhs.kind]
+    return mkNode ``calcStep #[← trExpr lhs, mkAtom ":=", ← trExpr rhs]
 
 mutual
 
   partial def trBlock : Block → M Syntax
     | ⟨_, none, none, #[]⟩ => do `(Parser.Tactic.tacticSeq| {})
     | ⟨_, none, none, tacs⟩ =>
       return mkNode ``Parser.Tactic.tacticSeq #[mkNode ``Parser.Tactic.tacticSeq1Indented #[
-        mkNullNode $ ← tacs.mapM fun tac => return mkGroupNode #[← trTactic tac.kind, mkNullNode]]]
+        mkNullNode $ ← tacs.mapM fun tac => return mkGroupNode #[← trTactic tac, mkNullNode]]]
     | ⟨_, cl, cfg, tacs⟩ => warn! "unsupported (TODO): block with cfg"
 
-  partial def trTactic : Tactic → M Syntax
+  partial def trTactic : Spanned Tactic → M Syntax := unspanningS fun
     | Tactic.block bl => do `(tactic| · ($(← trBlock bl):tacticSeq))
-    | Tactic.by tac => do `(tactic| · $(← trTactic tac.kind):tactic)
+    | Tactic.by tac => do `(tactic| · $(← trTactic tac):tactic)
     | Tactic.«;» tacs => do
       let rec build (i : Nat) (lhs : Syntax) : M Syntax :=
         if h : i < tacs.size then do
-          match ← trTacticOrList (tacs.get ⟨i, h⟩).kind with
+          match ← trTacticOrList (tacs.get ⟨i, h⟩) with
           | Sum.inl tac => `(tactic| $lhs <;> $(← build (i+1) tac))
           | Sum.inr tacs => build (i+1) (← `(tactic| $lhs <;> [$tacs,*]))
         else pure lhs
-      build 1 (← trTactic tacs[0].kind)
+      build 1 (← trTactic tacs[0])
     | Tactic.«<|>» tacs => do
-      `(tactic| first $[| $(← tacs.mapM fun tac => trTactic tac.kind):tactic]*)
+      `(tactic| first $[| $(← tacs.mapM fun tac => trTactic tac):tactic]*)
     | Tactic.«[]» tacs => warn! "unsupported (impossible)"
     | Tactic.exact_shortcut ⟨_, Expr.calc args⟩ => do
       `(tactic| calc $(← trCalcArgs args)*)
-    | Tactic.exact_shortcut e => do `(tactic| exact $(← trExpr e.kind))
+    | Tactic.exact_shortcut e => do `(tactic| exact $(← trExpr e))
     | Tactic.expr e =>
       match e.kind.unparen with
       | Expr.«`[]» tacs => trIdTactic ⟨true, none, none, tacs⟩
-      | e => do
+      | _ => do
```

Comment:
this is a behavior change, you aren't using the `unparen` result anymore. Perhaps `unparen` should be modified to be `Spanned Expr -> Spanned Expr` instead of `Expr -> Expr`.

Reply:
Okay, I've added a new unparen function for Spanned Expr.